coderd/database/custom_queries.go

```diff
@@ -0,0 +1,61 @@
+package database
+
+import (
+	"context"
+	"fmt"
+
+	"golang.org/x/xerrors"
+
+	"github.com/coder/coder/coderd/rbac"
+
+	"github.com/lib/pq"
+)
+
+// AuthorizedGetWorkspaces returns all workspaces that the user is authorized to access.
+func (q *sqlQuerier) AuthorizedGetWorkspaces(ctx context.Context, arg GetWorkspacesParams, authorizedFilter rbac.AuthorizeFilter) ([]Workspace, error) {
```

Comment:
I wish we could add hooks into SQLc to do this. Kinda sucks to have to copy the original function and call it like this.